go/runtime/bundle/discovery.go

```diff
@@ -515,6 +524,78 @@ func (d *Discovery) copyBundle(src string) error {
 	return nil
 }
 
+// cleanStaleExplodedBundles removes regular and detached bundles exploded bundle subdir,
+// for the runtimes no longer present in the configuration.
+func (d *Discovery) cleanStaleExplodedBundles(ctx context.Context) {
+	d.logger.Info("removing exploded (regular and detached) bundles for runtimes no longer present in the config")
+
+	cleanDir := func(bundlesDir string) {
+		entries, err := os.ReadDir(bundlesDir)
+		if err != nil {
+			if os.IsNotExist(err) {
+				return
+			}
+			d.logger.Error("error reading bundles dir",
```

Comment:
```suggestion
			d.logger.Error("failed to read bundles dir",
```
We usually start with `failed to` and not `error` (you already know this is error from error log).